tools/install.sh

```diff
@@ -1,116 +1,151 @@
+#!/bin/sh
+set -e
+#
+# This script should be run via curl:
+#   sh -c "$(curl -fsSL https://raw.githubusercontent.com/robbyrussell/oh-my-zsh/master/tools/install.sh)"
+# or wget:
+#   sh -c "$(wget -qO- https://raw.githubusercontent.com/robbyrussell/oh-my-zsh/master/tools/install.sh)"
+#
+# As an alternative, you can download the install script separately and
+# run it afterwards with `sh install.sh'
+#
+
+# Default location
+ZSH=${ZSH:-~/.oh-my-zsh}
+
+command_exists() {
+	command -v "$@" >/dev/null 2>&1
+}
+
+# Set up color sequences
+setup_color() {
+	if command_exists tput; then
+		ncolors=$(tput colors)
+	else
+		ncolors=0
+	fi
+
+	# Only use colors if connected to a terminal that supports them
+	if [ -t 1 ] && [ $ncolors -ge 8 ]; then
+		RED="$(tput setaf 1)"
+		GREEN="$(tput setaf 2)"
+		YELLOW="$(tput setaf 3)"
+		BLUE="$(tput setaf 4)"
+		BOLD="$(tput bold)"
+		NORMAL="$(tput sgr0)"
+	else
+		RED=""
+		GREEN=""
+		YELLOW=""
+		BLUE=""
+		BOLD=""
+		NORMAL=""
+	fi
+}
+
+setup_ohmyzsh() {
+	# Prevent the cloned repository from having insecure permissions. Failing to do
+	# so causes compinit() calls to fail with "command not found: compdef" errors
+	# for users with insecure umasks (e.g., "002", allowing group writability). Note
+	# that this will be ignored under Cygwin by default, as Windows ACLs take
+	# precedence over umasks except for filesystems mounted with option "noacl".
+	umask g-w,o-w
+
+	echo "${BLUE}Cloning Oh My Zsh...${NORMAL}"
+
+	command_exists git || {
+		echo "Error: git is not installed"
+		exit 1
+	}
+
+	if [ "$OSTYPE" = cygwin ] && git --version | grep -q msysgit; then
+		cat <<-EOF
+			Error: Windows/MSYS Git is not supported on Cygwin
+			Error: Make sure the Cygwin git package is installed and is first on the $PATH
+		EOF
+		exit 1
+	fi
+
+	git clone --depth=1 https://github.com/robbyrussell/oh-my-zsh.git $ZSH || {
+		echo "Error: git clone of oh-my-zsh repo failed"
+		exit 1
+	}
+}
+
+setup_zshrc() {
+	echo "${BLUE}Looking for an existing zsh config...${NORMAL}"
+	if [ -f ~/.zshrc ] || [ -h ~/.zshrc ]; then
+		echo "${YELLOW}Found ~/.zshrc.${GREEN} Backing up to ~/.zshrc.pre-oh-my-zsh.${NORMAL}"
+		mv ~/.zshrc ~/.zshrc.pre-oh-my-zsh
+	fi
+
+	echo "${BLUE}Using the Oh My Zsh template file and adding it to ~/.zshrc.${NORMAL}"
+
+	cp $ZSH/templates/zshrc.zsh-template ~/.zshrc
+	sed "/^export ZSH=/ c\\
+	export ZSH=$ZSH
+	" ~/.zshrc > ~/.zshrc-omztemp
+	mv -f ~/.zshrc-omztemp ~/.zshrc
+
+setup_shell() {
+	# If this user's login shell is not already "zsh", attempt to switch.
+	TEST_CURRENT_SHELL=$(expr "$SHELL" : '.*/\(.*\)')
+	if [ "$TEST_CURRENT_SHELL" != "zsh" ]; then
+		# If this platform provides a "chsh" command (not Cygwin), do it, man!
+		if command_exists chsh; then
```

Comment:
This assumes that `/etc/shells` exists if `chsh` exists, which is not the case on all systems (Android).

Could you either also check for `/etc/shells` here, or try `which zsh` to find the path before trying `$(grep /zsh$ /etc/shells | tail -1)` inside the if clause?

Reply:
#6398 should fix this